core/server/utils/image-size.js

```diff
@@ -165,27 +170,43 @@ getImageSizeFromUrl = function getImageSizeFromUrl(imagePath) {
  * @returns {object} imageObject or error
  */
 getImageSizeFromFilePath = function getImageSizeFromFilePath(imagePath) {
+    var filePath;
+
     imagePath = utils.url.urlFor('image', {image: imagePath}, true);
-    imageObject.url = imagePath;
 
-    imagePath = storageUtils.getLocalFileStoragePath(imagePath);
+    // get the storage readable filePath
+    filePath = storageUtils.getLocalFileStoragePath(imagePath);
 
     return storage.getStorage()
-        .read({path: imagePath})
+        .read({path: filePath})
         .then(function readFile(buf) {
-            debug('Image fetched (storage):', imagePath);
+            debug('Image fetched (storage):', filePath);
 
             return fetchDimensionsFromBuffer({
                 buffer: buf,
+                // we need to return the URL that's accessible for network requests as this imagePath
+                // value will be used as the URL for structured data
                 imagePath: imagePath
             });
-        })
-        .catch(function (err) {
+        }).catch({code: 'ENOENT'}, function (err) {
+            return Promise.reject(new errors.NotFoundError({
+                message: err.message,
+                code: 'IMAGE_SIZE',
+                err: err,
+                context: {
```

Comment:
@ErisDS I don't know if this makes sense, but I thought it would be useful information to provide the passed in filepath along with the path we used to read from the storage?

Reply:
`context` is for users and must be a string. If you want to provide extra details along with the error for debugging purposes, please use `errorDetails: {}`